src/items/use-declarations.md

```diff
@@ -442,6 +449,111 @@ fn main() {
 }
 ```
 
+r[names.resolution.early.imports.errors.ambiguity.builtin-attr]
+* it is an error to have a user defined attribute or derive macro with the same name as a builtin attribute (e.g. inline)
+    * I think we may special case this one and allow certain kinds of
+      ambiguities where the builtin-attr is shadowed by a user attribute (not
+      sure if this actually exists or is just proposed, TODO investigate)
+    * TODO example? This ones harder to do concisely afaik
+r[names.resolution.early.imports.errors.ambiguity.derivehelper]
+* derive helpers used before their associated derive may not shadow other attributes or other derive helpers that are otherwise in scope after their derive
```

Comment:
This is something that exists under a deprecation lint in the final stage of its lifecycle (https://github.com/rust-lang/rust/issues/79202), and hopefully will be removed next year.
Not sure if things like this are supposed to be documented in the reference.

Reply:
ive gone ahead and moved this to be in the proc macro helper attribute section and setup a first pass attempt at documenting the intended behavior as I understand it, the admonition about deprecation, and an example.